mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/NullSavingStrategyValidation.java

```diff
@@ -0,0 +1,246 @@
+   *     (including the default one)
+   */
+  boolean hasDoNotSetOnAnyLevel(
+      List<ExecutableElement> methodElements, TypeElement daoLevelInterface) {
+    boolean hasDoNotSetStatement =
+        methodElements
+            .stream()
+            .anyMatch(
+                v ->
+                    updateHasDoNotSet(v, false)
+                        || insertHasDoNotSet(v, false)
+                        || setEntityHasDoNotSet(v, false)
+                        || queryHasDoNotSet(v, false));
+
+    boolean hasDoNotSetStatementSetExplicitly =
+        methodElements
+            .stream()
+            .anyMatch(
+                v ->
+                    updateHasDoNotSet(v, true)
+                        || insertHasDoNotSet(v, true)
+                        || setEntityHasDoNotSet(v, true)
+                        || queryHasDoNotSet(v, true));
+
+    boolean hasSetToNullStatementSetExplicitlyForAll =
+        methodElements
+            .stream()
+            .filter(this::isOperationWithNullSavingStrategy)
+            .allMatch(
+                v ->
+                    updateHasSetToNullExplicitly(v)
+                        || insertHasSetToNullExplicitly(v)
+                        || setEntitySetToNullExplicitly(v)
+                        || queryHasSetToNullExplicitly(v));
+
+    // if DAO level SET_TO_NULL check all underlying annotations for explicit set to DO_NOT_SET
+    // (they may override it)
+    if (hasDaoLevelSetToNullStatement(daoLevelInterface) && hasDoNotSetStatementSetExplicitly) {
+      return true;
+      // if DAO level DO_NOT_SET check if all underlying override it explicitly to SET_TO_NULL
+    } else if (hasDaoLevelDoNotSetStatement(daoLevelInterface)
+        && !hasSetToNullStatementSetExplicitlyForAll) {
+      return true;
+      // if DAO level annotation do not present, check method level strategy
+      // (including the default one)
+    } else {
+      return hasDaoLevelUnset(daoLevelInterface) && hasDoNotSetStatement;
+    }
+  }
+
+  private boolean hasDaoLevelDoNotSetStatement(TypeElement interfaceElement) {
+    DefaultNullSavingStrategy annotation =
+        interfaceElement.getAnnotation(DefaultNullSavingStrategy.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+    }
+    return false;
+  }
+
+  private boolean hasDaoLevelSetToNullStatement(TypeElement interfaceElement) {
+    DefaultNullSavingStrategy annotation =
+        interfaceElement.getAnnotation(DefaultNullSavingStrategy.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL;
+    }
+    return false;
+  }
+
+  private boolean hasDaoLevelUnset(TypeElement interfaceElement) {
+    return interfaceElement.getAnnotation(DefaultNullSavingStrategy.class) == null;
+  }
+
+  private boolean queryHasDoNotSet(ExecutableElement v, boolean explicitSet) {
```

Comment:
There is a lot of boilerplate here between `queryHasDoNotSet`, `setEntityHasDoNotSet`, `insertHasDoNotSet` and `updateHasDoNotSet` that could probably be factored into a common method, for example you could do something like this:

```java
  private <A extends Annotation> boolean hasDoNotSet(Class<A> clazz,
                              Function<A, NullSavingStrategy> extractor,
                              ExecutableElement v,
                              boolean explicitSet) {
    A annotation = v.getAnnotation(clazz);
    if (annotation != null) {
      NullSavingStrategy strategy = extractor.apply(annotation);
      if (explicitSet) {
        return strategy == NullSavingStrategy.DO_NOT_SET
          && nullSavingStrategyExplicitlySet(v, clazz);
      } else {
        return strategy == NullSavingStrategy.DO_NOT_SET;
      }
    }
    return false;
  }

  private boolean queryHasDoNotSet(ExecutableElement v, boolean explicitSet) {
    return hasDoNotSet(Query.class, Query::nullSavingStrategy, v, explicitSet);
  }
```

Reply:
I was also thinking about it - I had an impression that sometimes we are favoring code readability over DRY but If that is your impression I am +1 on this as well.
Implemented in https://github.com/datastax/java-driver/pull/1246/commits/0e99daa6a413387ae8478a01b5fd88c463edbc4f